matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/store/db/RealmCryptoStoreMigration.kt

```diff
@@ -57,6 +58,29 @@ internal class RealmCryptoStoreMigration @Inject constructor(private val crossSi
         const val CRYPTO_STORE_SCHEMA_VERSION = 11L
     }
 
+    private fun RealmObjectSchema.addFieldIfNotExists(fieldName: String, fieldType: Class<*>): RealmObjectSchema {
+        if (!hasField(fieldName)) {
+            addField(fieldName, fieldType)
+        }
+        return this
+    }
+
+    private fun RealmObjectSchema.removeFieldIfExists(fieldName: String): RealmObjectSchema {
+        if (hasField(fieldName)) {
+            removeField(fieldName)
+        }
+        return this
+    }
+
+    private fun RealmObjectSchema.setRequiredIfNotAlready(fieldName: String, isRequired: Boolean): RealmObjectSchema {
+        if (isRequired && !isRequired(fieldName)) {
+            setRequired(fieldName, true)
+        } else if (!isRequired && isRequired(fieldName)) {
+            setRequired(fieldName, false)
+        }
```

Comment:
Could be simplified by
`if (isRequired != isRequired(fieldName) { setRequired(filedName, isRequired) }`

Reply:
Nice, done.